xarray/conventions.py

```diff
@@ -837,11 +838,12 @@ def decode_cf_variables(variables, attributes, concat_characters=True,
 
     def stackable(dim):
         # figure out if a dimension can be concatenated over
-        if dim in variables:
+        if dim in variables and dim not in concat_dimensions:
```

Comment:
Apologies for the delayed response, @shoyer...

I guess my concern was making sure this still worked as expected when opening up netCDF files (e.g. it goes through all the checks and concatenates as it should, etc.), so I didn't want to change _too_ much, but I can see how that looks a bit sloppy.

Gave it a bit of thought, maybe something like this would work:

```
    def stackable(dim):
        # figure out if a dimension can be concatenated over
        # concat_dimensions is turned into a set (like drop_variables) before
        # this function is used.
        if concat_dimensions:
            return dim in concat_dimensions
        else:
            if dim in variables:
                return False
            for v in dimensions_used_by[dim]:
                if v.dtype.kind != 'S' or dim != v.dims[-1]:
                    return False
        return True
```

...or maybe I'm misunderstanding what you're suggesting.

Also: I wonder if `concat_dimensions` should be added as a kwarg in `open_dataset`? If the user knows the dimension names to concatenate before hand, I could see that functionality being useful.